### ISSUE TITLE
`USE_BIG_DECIMAL_FOR_FLOATS` feature not working when using `JSON.treeFrom()`

### ISSUE THREAD

**Issue by jvdsandt** [FasterXML/jackson-jr#90]
This test fails because the USE_BIG_DECIMAL_FOR_FLOATS feature is ignored and Double objects are stiill used.

    @Test
    public void testParseDecimals() throws Exception {

        JSON json = JSON.builder()
                .treeCodec(new JacksonJrsTreeCodec())
                .enable(JSON.Feature.USE_BIG_DECIMAL_FOR_FLOATS)
                .build();

        String input = "[1.1]";

        TreeNode node = json.treeFrom(input);
        TreeNode elemNode = node.get(0);

        assertTrue(elemNode instanceof JrsNumber);
        assertTrue(((JrsNumber) elemNode).getValue() instanceof BigDecimal);
    }

**Comment by cowtowncoder** [comment]
Thank you for reporting this. I added a failing test to reproduce the problem.

Unfortunately `JacksonJrsTreeCodec` has no access to feature flags of `JSON`, so it cannot really determine this setting at this point. To solve the problem we'd probably have to do one of:

1. Figure out how to let codec access feature settings at time of registration (I am  not sure it is quite possible to have fully dynamic link for new `JSON` instances)
2. Add separate setting(s) for codec to be configured to do coercion.

**Comment by cowtowncoder** [comment]
Fixed for 2.17.1.

NOTE: **MUST** register codec as extension like so:

```
        JSON json = JSON.builder()
                .enable(JSON.Feature.USE_BIG_DECIMAL_FOR_FLOATS)
                .register(new JrSimpleTreeExtension())
                .build();
```

for things to work; registering directly as `TreeCodec` will not work.
